### ISSUE TITLE
API is not showing updated Syscollector data

### ISSUE THREAD

**Issue by dbray925** [wazuh/wazuh#23847]
|Wazuh version|Component|Install type|Platform|
|---|---|---|---|---|
| 4.7.5-1 | Syscollector | RPM | Rocky Linux 8.9 |

## Description
The API is not showing updated Syscollector (inventory) information, but the SQLite database files are.

Checked the debug logs on the agent, and nothing helpful:
2024/06/03 14:22:48 wazuh-modulesd:syscollector: INFO: Module started.
2024/06/03 14:22:48 wazuh-modulesd:syscollector: INFO: Starting evaluation.
2024/06/03 14:22:48 wazuh-modulesd:syscollector: INFO: Evaluation finished.

In order to check the local.db file, I had to install sqlite3 which is a great test to see if that package is showing up in the inventory, which it _now_ is on the local agent database:
`sudo sqlite3 /var/ossec/queue/syscollector/db/local.db "SELECT name,version,DATETIME(install_time,'unixepoch') as install_time FROM dbsync_packages WHERE name LIKE 'sql%';"
`
> sqlite|3.26.0-19.el8_9|2024-06-03 13:35:36
> sqlite-libs|3.26.0-19.el8_9|2024-04-12 12:59:39

I then checked the Wazuh server agent ID database, and the same shows up there:
`sudo sqlite3 /var/ossec/queue/db/291.db "SELECT scan_time,DATETIME(install_time,'unixepoch') as install_time,name,version FROM sys_programs WHERE name LIKE 'sqlite%' LIMIT 2"`
> 2024/05/30 14:44:42|2024-04-12 12:59:39|sqlite-libs|3.26.0-19.el8_9
> 2024/06/03 13:52:37|2024-06-03 13:35:36|sqlite|3.26.0-19.el8_9

However, when checking the API, it only shows the original libs package:
`curl -s -k -X GET "https://localhost:55000/syscollector/$AGENTID/$PROPERTY?pretty=true" -H "Authorization: Bearer $TOKEN"  | grep -e '"name":' | grep sqlite`
>             "name": "sqlite-libs",

## Checks

The following elements have been updated or reviewed (should also be checked if no modification is required):
- [ ] Tests (unit tests, API integration tests).
- [ ] Changelog.
- [ ] Documentation.
- [ ] Integration test mapping (using `api/test/integration/mapping/_test_mapping.py`).

**Comment by MiguelazoDS** [comment]
Hi @dbray925, 

I've never faced something similar to that, it is strange because the API will query the databases to respond that, there's no other place to find the packages installed in an agent, the only thing that I'm thinking is that you have more packages than the limit and just because of luck, the sqlite-libs package is among that limit, and the sqlite package didn't fit. 

Could you confirm that? 

https://documentation.wazuh.com/current/user-manual/api/reference.html#operation/api.controllers.syscollector_controller.get_packages_info

Regards!

**Comment by dbray925** [comment]
Sorry, not really sure what you are asking me to confirm. My example above, while using bash variables, is already showing the results of the API GET syscollector/$AGENTID/packages

**Comment by MiguelazoDS** [comment]
How many packages did you get as a response?

**Comment by dbray925** [comment]
My apologies, thanks for the clarification.

I've done various agent checks:
`curl -s -k -X GET "https://localhost:55000/syscollector/$AGENTID/$PROPERTY?pretty=true" -H "Authorization: Bearer $TOKEN" | grep -e '"name"' | wc -l`

...all keep coming back with: 500

I think you might be onto something. _All_ agents keep coming back with 500. I'd have to assume you're right, and it just happen to hit that limit. Very odd!

**Comment by MiguelazoDS** [comment]
Yeah, that's it. Just by coincidence, the package SQLite is out of that range of 500 elements. 

Let's try this 


```console
curl -s -k -X GET "https://localhost:55000/syscollector/$AGENTID/$PROPERTY?limit=2000?pretty=true" -H "Authorization: Bearer $TOKEN"  | grep -e '"name":' | grep sqlite
```

**Comment by dbray925** [comment]
Small adjustment to the parameters: & instead of ?, but that is working now. Thank you!  Wow, never new there was a limit. 
`curl -s -k -X GET "https://localhost:55000/syscollector/$AGENTID/$PROPERTY?limit=2000&pretty=true" -H "Authorization: Bearer $TOKEN"  | grep -e '"name":' | wc -l`

...this is now showing 565 for some, 529 for others, etc.

**Comment by dbray925** [comment]
@MiguelazoDS thank you again for the help with this, and apologies for wasting a Git issue on something that is clearly documented: https://documentation.wazuh.com/current/user-manual/api/getting-started.html#basic-concepts

> The limit parameter accepts up to 100000 items, although it is recommended not to exceed the default value (500 items).

I should've caught that.

**Comment by MiguelazoDS** [comment]
Yeah "&" is correct to separate the parameters, my bad. Glad we found the issue, it was weird because API uses the databases to respond. 

GitHub issues are intended for this, regards!
